drivers/util.py

```diff
@@ -1098,24 +1097,30 @@ def dom0_disks():
     return disks
 
 
-def set_scheduler_sysfs_node(node, str):
-    """Set the scheduler for a sysfs node (e.g. '/sys/block/sda')"""
+def set_scheduler_sysfs_node(node, scheds):
+    """
+    Set the scheduler for a sysfs node (e.g. '/sys/block/sda')
+    according to prioritized list schedulers
+    Try to set the first item, then fall back to the next on failure
+    """
 
     path = os.path.join(node, "queue", "scheduler")
     if not os.path.exists(path):
         SMlog("no path %s" % path)
         return
-    try:
-        f = open(path, 'w')
-        f.write("%s\n" % str)
-        f.close()
-        SMlog("Set scheduler to [%s] on [%s]" % (str, node))
-    except:
-        SMlog("Error setting scheduler to [%s] on [%s]" % (str, node))
-        pass
+    for sched in scheds:
+        try:
+            with open(path, 'w') as file:
+                file.write("%s\n" % sched)
+            SMlog("Set scheduler to [%s] on [%s]" % (sched, node))
+            return
+        #pylint: disable=broad-except
+        except Exception as err:
```

Comment:
We should be able to be more specific about the exception type here (and similarly the one in SR.py)

Reply:
<also for SR.py>

These are actually the re-write of the original code, which catch all the exception.
BTW, I do think we should catch and ignore all the exceptions and try the next one on failure.
But if the specific exception is really preferred, I can update to it. @MarkSymsCtx  
```
-    try:
-        f = open(path, 'w')
-        f.write("%s\n" % str)
-        f.close()
-        SMlog("Set scheduler to [%s] on [%s]" % (str, node))
-    except:
-        SMlog("Error setting scheduler to [%s] on [%s]" % (str, node))
-        pass
```